### PR TITLE
MM-49703: Bump to Go 1.19

### PR DESCRIPTION
**Pull request by agnivade**
Also fix some gofmt issues

https://mattermost.atlassian.net/browse/MM-49703

### CODE REVIEW

**Comment by agnivade** [comment]
There is one more change pending here. I need to update this line `IMAGE_BUILDER: $CI_REGISTRY/mattermost/ci/images/builder:go-1.18.1-node-16.15.0-1 ` in .gitlab-ci.yml

**Review by isacikgoz — APPROVED** [review]

**Comment by agnivade** [comment]
ping @wiggin77

File: server/api/auth.go [hunk 1]
```diff
@@ -383,12 +383,13 @@ func (a *API) attachSession(handler func(w http.ResponseWriter, r *http.Request)
 
 		authService := session.AuthService
 		if authService != a.authService {
-			a.logger.Error(`Session authService mismatch`,
+			msg := `Session authService mismatch`
```
Comment (phoinixgrr) [inline server/api/auth.go:386]: Are these changes related to the GoLang update?

File: server/api/auth.go [hunk 2]
```diff
@@ -383,12 +383,13 @@ func (a *API) attachSession(handler func(w http.ResponseWriter, r *http.Request)
...
+			a.logger.Error(msg,
 				mlog.String("sessionID", session.ID),
 				mlog.String("want", a.authService),
 				mlog.String("got", authService),
 			)
-			a.errorResponse(w, r, model.NewErrUnauthorized(err.Error()))
+			a.errorResponse(w, r, model.NewErrUnauthorized(msg))
```
Comment (phoinixgrr) [inline server/api/auth.go:392]: Are these changes related to the GoLang update?

File: server/api/members.go [hunk 3]
```diff
@@ -146,7 +146,7 @@ func (a *API) handleAddMember(w http.ResponseWriter, r *http.Request) {
 	}
 
 	if reqBoardMember.UserID == "" {
-		a.errorResponse(w, r, model.NewErrBadRequest(err.Error()))
+		a.errorResponse(w, r, model.NewErrBadRequest("empty userID"))
```
Comment (phoinixgrr) [inline server/api/members.go:149]: Are these changes related to the GoLang update?

File: server/app/insights.go [hunk 4]
```diff
@@ -19,7 +19,7 @@ func (a *App) GetTeamBoardsInsights(userID string, teamID string, opts *mmModel.
 	if err != nil {
 		return nil, err
 	}
-	return a.store.GetTeamBoardsInsights(teamID, userID, opts.StartUnixMilli, opts.Page*opts.PerPage, opts.PerPage, boardIDs)
+	return a.store.GetTeamBoardsInsights(teamID, opts.StartUnixMilli, opts.Page*opts.PerPage, opts.PerPage, boardIDs)
```
Comment (phoinixgrr) [inline server/app/insights.go:22]: Are these changes related to the GoLang update?

File: server/app/insights_test.go [hunk 5]
```diff
@@ -51,7 +51,7 @@ func TestGetTeamAndUserBoardsInsights(t *testing.T) {
 		th.Store.EXPECT().GetUserByID("user-id").Return(fakeUser, nil).AnyTimes()
 		th.Store.EXPECT().GetBoardsForUserAndTeam("user-id", "team-id", true).Return(mockInsightsBoards, nil).AnyTimes()
 		th.Store.EXPECT().
-			GetTeamBoardsInsights("team-id", "user-id", int64(0), 0, 10, []string{"mock-user-workspace-id"}).
+			GetTeamBoardsInsights("team-id", int64(0), 0, 10, []string{"mock-user-workspace-id"}).
```
Comment (phoinixgrr) [inline server/app/insights_test.go:54]: Are these changes related to the GoLang update?

File: server/app/insights_test.go [hunk 6]
```diff
@@ -74,7 +74,7 @@ func TestGetTeamAndUserBoardsInsights(t *testing.T) {
...
```
Comment (phoinixgrr) [inline server/app/insights_test.go:77]: Are these changes related to the GoLang update?

File: server/main/main.go [hunk 7]
```diff
@@ -122,7 +122,7 @@ func main() {
 	if pDBConfig != nil && len(*pDBConfig) > 0 {
 		config.DBConfigString = *pDBConfig
 		// Don't echo, as the confix string may contain passwords
-		logger.Info("DBConfigString overriden from commandline")
+		logger.Info("DBConfigString overridden from commandline")
```
Comment (phoinixgrr) [inline server/main/main.go:125]: Are these changes related to the GoLang update?

File: server/services/store/sqlstore/board_insights.go [hunk 8]
```diff
@@ -14,7 +14,7 @@ import (
 	"github.com/mattermost/mattermost-server/v6/shared/mlog"
 )
 
-func (s *SQLStore) getTeamBoardsInsights(db sq.BaseRunner, teamID string, userID string, since int64, offset int, limit int, boardIDs []string) (*model.BoardInsightsList, error) {
+func (s *SQLStore) getTeamBoardsInsights(db sq.BaseRunner, teamID string, since int64, offset int, limit int, boardIDs []string) (*model.BoardInsightsList, error) {
```
Comment (phoinixgrr) [inline server/services/store/sqlstore/board_insights.go:17]: Are these changes related to the GoLang update?

File: server/services/store/sqlstore/sqlstore.go [hunk 9]
```diff
@@ -185,7 +185,7 @@ func (s *SQLStore) getChannel(db sq.BaseRunner, teamID, channel string) (*mmMode
 	return nil, store.NewNotSupportedError("get channel not supported on standalone mode")
 }
 
-func (s *SQLStore) dBVersion(db sq.BaseRunner) string {
+func (s *SQLStore) dBVersion() string {
```
Comment (phoinixgrr) [inline server/services/store/sqlstore/sqlstore.go:188]: Are these changes related to the GoLang update?

File: server/services/store/storetests/board_insights.go [hunk 10]
```diff
@@ -72,7 +72,7 @@ func getBoardsInsightsTest(t *testing.T, store store.Store) {
 	boardsUser2, _ := store.GetBoardsForUserAndTeam(testInsightsUserID1, testTeamID, true)
 	t.Run("team insights", func(t *testing.T) {
 		boardIDs := []string{boardsUser1[0].ID, boardsUser1[1].ID, boardsUser1[2].ID}
-		topTeamBoards, err := store.GetTeamBoardsInsights(testTeamID, testUserID,
+		topTeamBoards, err := store.GetTeamBoardsInsights(testTeamID,
```
Comment (phoinixgrr) [inline server/services/store/storetests/board_insights.go:75]: Are these changes related to the GoLang update?

File: server/ws/plugin_adapter.go [hunk 11]
```diff
@@ -422,7 +422,7 @@ func (pa *PluginAdapter) sendTeamMessage(event, teamID string, payload map[strin
 			EnsureUsers: ensureUserIDs,
 		}
 
-		pa.sendMessageToCluster("websocket_message", clusterMessage)
+		pa.sendMessageToCluster(clusterMessage)
```
Comment (phoinixgrr) [inline server/ws/plugin_adapter.go:425]: Are these changes related to the GoLang update?

File: server/ws/plugin_adapter.go [hunk 12]
```diff
@@ -447,7 +447,7 @@ func (pa *PluginAdapter) sendBoardMessage(teamID, boardID string, payload map[st
...
```
Comment (phoinixgrr) [inline server/ws/plugin_adapter.go:450]: Are these changes related to the GoLang update?

File: server/ws/plugin_adapter.go [hunk 13]
```diff
@@ -490,7 +490,7 @@ func (pa *PluginAdapter) BroadcastCategoryChange(category model.Category) {
 			UserID:  category.UserID,
 		}
 
-		pa.sendMessageToCluster("websocket_message", clusterMessage)
+		pa.sendMessageToCluster(clusterMessage)
```
Comment (phoinixgrr) [inline server/ws/plugin_adapter.go:493]: Are these changes related to the GoLang update?

File: server/ws/plugin_adapter.go [hunk 14]
```diff
@@ -514,7 +514,7 @@ func (pa *PluginAdapter) BroadcastCategoryReorder(teamID, userID string, categor
 			UserID:  userID,
 		}
 
-		pa.sendMessageToCluster("websocket_message", clusterMessage)
+		pa.sendMessageToCluster(clusterMessage)
```
Comment (phoinixgrr) [inline server/ws/plugin_adapter.go:517]: Are these changes related to the GoLang update?

File: server/ws/plugin_adapter.go [hunk 15]
```diff
@@ -540,7 +540,7 @@ func (pa *PluginAdapter) BroadcastCategoryBoardsReorder(teamID, userID, category
...
```
Comment (phoinixgrr) [inline server/ws/plugin_adapter.go:543]: Are these changes related to the GoLang update?

File: server/ws/plugin_adapter.go [hunk 16]
```diff
@@ -568,7 +568,7 @@ func (pa *PluginAdapter) BroadcastCategoryBoardChange(teamID, userID string, boa
...
```
Comment (phoinixgrr) [inline server/ws/plugin_adapter.go:571]: Are these changes related to the GoLang update?

File: server/ws/plugin_adapter_cluster.go [hunk 17]
```diff
@@ -15,7 +15,8 @@ type ClusterMessage struct {
 	EnsureUsers []string
 }
 
-func (pa *PluginAdapter) sendMessageToCluster(id string, clusterMessage *ClusterMessage) {
+func (pa *PluginAdapter) sendMessageToCluster(clusterMessage *ClusterMessage) {
+	const id = "websocket_message"
```
Comment (phoinixgrr) [inline server/ws/plugin_adapter_cluster.go:19]: Are these changes related to the GoLang update?

**Comment by agnivade** [comment]
All of those changes are partially related to the Go update. The issue is that the golangci version we use doesn't work with 1.19. So we had to bump golangci to 1.50.1. And it caught some additional bugs. Those bugs are fixed as well with this PR.

**Comment by phoinixgrr** [comment]
> All of those changes are partially related to the Go update. The issue is that the golangci version we use doesn't work with 1.19. So we had to bump golangci to 1.50.1. And it caught some additional bugs. Those bugs are fixed as well with this PR.

Approving the changes concerning the Golang Update.  Not able to review the bug fixes 
:shipit:

**Review by phoinixgrr — APPROVED** [review]

**Review by wiggin77 — APPROVED** [review]
LGTM :`  Thanks @agnivade for fixing the new linter errors.
